appservice/src/intTest/groovy/nu/yona/server/VpnConnectionStatusEventTest.groovy

```diff
@@ -0,0 +1,163 @@
+
+import groovy.json.*
+import nu.yona.server.test.User
+
+class VpnConnectionStatusEventTest extends AbstractAppServiceIntegrationTest
+{
+	def 'Richard posts initial VPN connected event: connection status updated and buddies informed'()
+	{
+		given:
+		def richardAndBob = addRichardAndBobAsBuddies()
+		User richard = richardAndBob.richard
+		User bob = richardAndBob.bob
+
+		when:
+		def response = richard.devices[0].postVpnStatus(appService, true)
+
+		then:
+		assertResponseStatusNoContent(response)
+
+		User richardAfterReload = appService.reloadUser(richard)
+		User bobAfterReload = appService.reloadUser(bob)
+		!richard.devices[0].vpnConnected
+		richardAfterReload.devices[0].vpnConnected
+		!bob.buddies[0].user.devices[0].vpnConnected
+		bobAfterReload.buddies[0].user.devices[0].vpnConnected
+
+		def bobMessagesAfterUpdate = appService.getMessages(bob)
+		assertResponseStatusOk(bobMessagesAfterUpdate)
+		def vpnConnectionStatusChangeMessages = bobMessagesAfterUpdate.responseData._embedded?."yona:messages".findAll{ it."@type" == "BuddyVpnConnectionStatusChangeMessage"}
+
+		vpnConnectionStatusChangeMessages.size() == 1
+		vpnConnectionStatusChangeMessages[0]._links.self != null
+		vpnConnectionStatusChangeMessages[0]._links."yona:user".href == bob.buddies[0].user.url
+		vpnConnectionStatusChangeMessages[0].message == "User connected VPN on device '${richard.devices[0].name}'"
+
+		cleanup:
+		appService.deleteUser(richard)
+		appService.deleteUser(bob)
+	}
+
+	def 'Richard posts a VPN disconnected event: connection status updated and buddies informed'()
+	{
+		given:
+		def richardAndBob = addRichardAndBobAsBuddies()
+		User richard = richardAndBob.richard
+		User bob = richardAndBob.bob
+		assertResponseStatusNoContent(richard.devices[0].postVpnStatus(appService, true))
+		richard = appService.reloadUser(richard)
+		bob = appService.reloadUser(bob)
+
+		when:
+		def response = richard.devices[0].postVpnStatus(appService, false)
+
+		then:
+		assertResponseStatusNoContent(response)
+
+		User richardAfterReload = appService.reloadUser(richard)
+		User bobAfterReload = appService.reloadUser(bob)
+		richard.devices[0].vpnConnected
+		!richardAfterReload.devices[0].vpnConnected
+		bob.buddies[0].user.devices[0].vpnConnected
+		!bobAfterReload.buddies[0].user.devices[0].vpnConnected
+
+		def bobMessagesAfterUpdate = appService.getMessages(bob)
+		assertResponseStatusOk(bobMessagesAfterUpdate)
+		def vpnConnectionStatusChangeMessages = bobMessagesAfterUpdate.responseData._embedded?."yona:messages".findAll{ it."@type" == "BuddyVpnConnectionStatusChangeMessage"}
+
+		vpnConnectionStatusChangeMessages.size() == 2
+		vpnConnectionStatusChangeMessages[0]._links.self != null
+		vpnConnectionStatusChangeMessages[0]._links."yona:user".href == bob.buddies[0].user.url
+		vpnConnectionStatusChangeMessages[0].message == "User disconnected VPN on device '${richard.devices[0].name}'"
+
+		vpnConnectionStatusChangeMessages[1]._links.self != null
```

Comment:
These asserts on the older message are less relevant for the test

Reply:
To me, it helps understanding the test: see there are two such message, and what they contain.